rust/src/world/yaml_loader.rs

```diff
@@ -0,0 +1,149 @@
+use super::error::SpanErr;
+use super::gltf::SuteraGltfObject;
+use godot::prelude::*;
+use serde::Deserialize;
+use serde_yaml::{Deserializer, Sequence, Value};
+use std::borrow::Cow;
+use std::fs::File;
+use std::io::prelude::*;
+use thiserror::Error;
+use tracing::instrument;
+
+#[instrument(name = "load_world", level = "trace")]
+pub fn load_world(
+    yaml_path: String,
+    parent_node: &mut Gd<Node>,
+) -> Result<(), SpanErr<WorldLoadingError>> {
+    let mut base_node = parent_node.clone();
+
+    //yamlファイルのpathを開く
+    let mut yaml_file = File::open(yaml_path)
+        .map_err(|e| SpanErr::from(WorldLoadingError::InvalidYamlPathErr(e.to_string())))?;
+    let mut contents = String::new();
+    yaml_file
+        .read_to_string(&mut contents)
+        .map_err(|e| SpanErr::from(WorldLoadingError::YamlFileOpenError(e.to_string())))?; //yamlファイルの内容をString型に
+
+    let value = Value::deserialize(Deserializer::from_str(&contents))  //yamlファイルをデシリアライズしてserdeのValue型(Enum型)を得る
+        .map_err(|e|{SpanErr::from(WorldLoadingError::SerdeYamlLoadingError(e.to_string()))})?;
```

Comment:
```suggestion
        .map_err(|e| SpanErr::from(WorldLoadingError::SerdeYamlLoadingError(e.to_string())))?;
```

↑の同じような処理と合わせる意味でこっちの方が良いかも

Reply:
修正しました！